### PR TITLE
Upgrade to scalajs-dom 2

### PR DESCRIPTION
**Pull request by rpiaggio**
Incudes upgrading to `Fetch`.

I'd love to make an `https4s-dom` module, but it doesn't support web sockets yet.

### CODE REVIEW

**Comment by rpiaggio** [comment]
> I'd love to make an `https4s-dom` module, but it doesn't support web sockets yet.

Actually, we can make a transactional-only backend. I'll work on that in a future PR.

File: scalajs/src/main/scala/clue/js/FetchJSBackend.scala [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+final class FetchJSBackend[F[_]: Async](method: FetchMethod) extends TransactionalBackend[F] {
+  def request(
+    uri:     Uri,
+    request: GraphQLRequest,
+    headers: Headers
+  ): F[String] =
+    Async[F].async_ { cb =>
+      val headersʹ = new FetchHeaders()
+      headers.headers.foreach(h => headersʹ.append(h.name.toString, h.value))
```
Comment (cquiroz) [inline scalajs/src/main/scala/clue/js/FetchJSBackend.scala:38]: I suppose you can pass an auth token in the headers

**Review by cquiroz — APPROVED** [review]
LGTM

**Comment by cquiroz** [comment]
> Actually, we can make a transactional-only backend. I'll work on that in a future PR.

Certainly we can, what do we get over a plain Fetch Backend?

File: scalajs/src/main/scala/clue/js/FetchJSBackend.scala (hunk 1 above)
Comment (rpiaggio) [reply scalajs/src/main/scala/clue/js/FetchJSBackend.scala:38]: Should be the case, same as before.

**Comment by rpiaggio** [comment]
> > Actually, we can make a transactional-only backend. I'll work on that in a future PR.
> 
> Certainly we can, what do we get over a plain Fetch Backend?

Full configuration with all Fetch options + timeout and cancellation.

**Comment by armanbilge** [comment]
I opened https://github.com/http4s/http4s/pull/5520 to try and get the ball rolling on a web socket client for http4s-dom. Since you have experience with this, would be really great to get your thoughts on the API design there.

I took a quick look at your JDK WS backend and it looks like you are only using the high-level API, which I think we should be able to support completely in http4s-dom. So then, maybe you can have a shared JVM/JS http4s backend for clue, that works with both the JDK client and the browser client :)

**Comment by rpiaggio** [comment]
That's right, we are only using the high level API. It would be great to have shared http4s backend. Thank you for the initiative.

**Pull request merged**